pysrc/whenever/__init__.pyi

```diff
@@ -40,8 +40,6 @@ __all__ = [
     "SUNDAY",
 ]
 
-Disambiguate = Literal["raise", "earlier", "later", "compatible"]
```

Comment:
Are you sure you want to remove this and add it to every method?
Imho this works well with every IDE.

Reply:
The possible values appear once you start typing, but the alias `Disambiguation` still appears in the signature.

<img width="483" alt="image" src="https://github.com/user-attachments/assets/5f22ba70-0bcd-4608-b685-549bb429eaa8">

Just having the literals is probably less confusing for the average user, who might at first glance think that `Disambiguation` is some class they've never heard of.

<img width="463" alt="image" src="https://github.com/user-attachments/assets/d2478302-dbda-40ec-998f-6fc7e0c3a457">

(screenshots from VSCode)

what do you think?

Reply:
From a user perspective the latter is obviously better. However repeating the definition every time is not very nice, either. I have no strong opinion on the issue and the argument you are making with vs code is a valid one.

What happens if you explicitly assign it a `TypeAlias`?
```python
Disambiguate: TypeAlias = Literal["raise", "earlier", "later", "compatible"]
```

If that doesn't change anything I think it's best to repeat the definition and maybe open a follow up issue for vs code?

Reply:
Using `TypeAlias` still gives the same problem.

Here's the thing though: I don't think VSCode (and the underlying pylance language server) are _wrong_: you could just as well make the case that the name of a type aliase is significant, and should be used in the signature.

In the end, repeating the same `Literal` may not be DRY, but I don't expect it to change much. If it ever does need to change, it's a simple search/replace away

Reply:
> I don't think VSCode (and the underlying pylance language server) are _wrong_: you could just as well make the case that the name of a type aliase is significant, and should be used in the signature. It just happens that in this case, we _prefer_ not to use the actual alias name.

I don't think so. Imho the type alias is only there so you don't have to repeat the type hints.
All my doc builds with Sphinx and PyCharm resolve them properly.
If the type name matters the correct thing would (probably) be to use `NewType`.

Reply:
> If the type name matters the correct thing would (probably) be to use `NewType`.

`NewType` behaves differently though!

> All my doc builds with Sphinx and PyCharm resolve them properly

Yes—although a [quick google](https://www.google.com/search?client=firefox-b-d&q=sphinx+show+type+alias) shows there isn't a consensus 🤔.

I'm marking this as "resolved" for now—since we just need to work with the tooling as it currently exists 🤷